### PR TITLE
[test] added timm models to test model zoo

### PR DESCRIPTION
**Pull request by FrankLeeeee**
## 📌 Checklist before creating the PR

- [x] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

part of efforts for #3049 

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

This PR added the model zoo for testing and documented the specifications. As a simple test, I have transformed the tracer testing on TIMM models so that it uses the latest kit.

## 💥 Checklist before requesting a review

- [x] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

**Comment by 1SAA** [comment]
Hi @FrankLeeeee 

Could you use this model list instead? It seems it has more models.

```
        tmm.beit_base_patch16_224,
        tmm.beitv2_base_patch16_224,
        tmm.cait_s24_224,
        tmm.coat_lite_mini,
        tmm.convit_base,
        tmm.deit3_base_patch16_224,
        tmm.dm_nfnet_f0,
        tmm.eca_nfnet_l0,
        tmm.efficientformer_l1,
        tmm.ese_vovnet19b_dw,
        tmm.gmixer_12_224,
        tmm.gmlp_b16_224,
        tmm.hardcorenas_a,
        tmm.hrnet_w18_small,
        tmm.inception_v3,
        tmm.mixer_b16_224,
        tmm.nf_ecaresnet101,
        tmm.nf_regnet_b0,
        tmm.regnetv_040,
        tmm.skresnet18,
        tmm.swin_base_patch4_window7_224,
        tmm.tnt_b_patch16_224,
        tmm.vgg11,
        tmm.vit_base_patch16_18x2_224,
        tmm.wide_resnet50_2,
```

**Comment by FrankLeeeee** [comment]
> Hi @FrankLeeeee
> 
> Could you use this model list instead? It seems it has more models. They used by @super-dainiu in the latest testing code.
> 
> ```
>         tmm.beit_base_patch16_224,
>         tmm.beitv2_base_patch16_224,
>         tmm.cait_s24_224,
>         tmm.coat_lite_mini,
>         tmm.convit_base,
>         tmm.deit3_base_patch16_224,
>         tmm.dm_nfnet_f0,
>         tmm.eca_nfnet_l0,
>         tmm.efficientformer_l1,
>         tmm.ese_vovnet19b_dw,
>         tmm.gmixer_12_224,
>         tmm.gmlp_b16_224,
>         tmm.hardcorenas_a,
>         tmm.hrnet_w18_small,
>         tmm.inception_v3,
>         tmm.mixer_b16_224,
>         tmm.nf_ecaresnet101,
>         tmm.nf_regnet_b0,
>         tmm.regnetv_040,
>         tmm.skresnet18,
>         tmm.swin_base_patch4_window7_224,
>         tmm.tnt_b_patch16_224,
>         tmm.vgg11,
>         tmm.vit_base_patch16_18x2_224,
>         tmm.wide_resnet50_2,
> ```

Ok cool.

File: tests/test_fx/test_tracer/test_timm_model/test_timm_model.py [hunk 1]
```diff
@@ -14,60 +15,47 @@ def trace_and_compare(model_cls, data, meta_args=None):
...
-    data = torch.rand(2, 3, 224, 224)
+    assert len(transformed_fx_out) == len(transformed_non_fx_out)
 
-    for model_cls in MODEL_LIST:
-        trace_and_compare(model_cls, data)
+    for key in transformed_fx_out.keys():
+        fx_output_val = transformed_fx_out[key]
+        non_fx_output_val = transformed_non_fx_out[key]
+        assert torch.allclose(fx_output_val, non_fx_output_val, atol=1e-5), \
+            f'{model.__class__.__name__} has inconsistent outputs, {fx_output_val} vs {non_fx_output_val}'
```
Comment (1SAA) [inline tests/test_fx/test_tracer/test_timm_model/test_timm_model.py:42]: It's better to have a function like `assert_dict(da, db, assert_fn)`, though not necessary.

**Review by 1SAA — APPROVED** [review]

File: tests/kit/model_zoo/registry.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    """
+    A registry to map model names to model and data generation functions.
+    """
+
+    def register(self,
+                 name: str,
+                 model_fn: Callable,
+                 data_gen_fn: Callable,
+                 output_transform_fn: Callable,
+                 model_attribute: ModelAttribute = None):
```
Comment (ver217) [inline tests/kit/model_zoo/registry.py:26]: How to set model init args? By this `model_attribute` or overwriting `model_fn`?

Comment (1SAA) [reply tests/kit/model_zoo/registry.py:26]: @FrankLeeeee I agree with @ver217. We should add config_kwargs for every type of model.

Comment (FrankLeeeee) [reply tests/kit/model_zoo/registry.py:26]: > 

`model_attribute` is used for logic checks, for example, you may want to differentiate models with/without control flow in tracing/auto parallel. If you want to set init args when registering. You can do:

```
model_fn = lambda: model_cls(arg1=x, arg2=y)
```

File: tests/test_fx/test_tracer/test_timm_model/test_timm_model.py (hunk 1 above)
Comment (FrankLeeeee) [reply tests/test_fx/test_tracer/test_timm_model/test_timm_model.py:42]: I can add that to `colossalai.testing`, but it would be in a separate PR.

**Pull request merged**